src/Microsoft.Health.Fhir.Api/Features/ActionResults/BaseActionResult.cs

```diff
@@ -0,0 +1,62 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Net;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Health.Fhir.Api.Features.ActionResults
+{
+    public abstract class BaseActionResult<T> : ActionResult
+    {
+        /// <summary>
+        /// Gets the payload associated with this action result.
+        /// </summary>
+        public T Payload { get; protected set; }
```

Comment:
nit: maybe 'Result'? Also, you should make this getter only and then have a `protected BaseActionResult(T result)` as opposed to have a protected setter.